src/Culling/babylon.boundingBox.ts

```diff
@@ -131,7 +135,7 @@
          * @returns current bounding box
          */
         public setWorldMatrix(matrix: Matrix): BoundingBox {
-            this._worldMatrix.copyFrom(matrix);
+            this._worldMatrix = matrix;

```

Comment:
Nope, you don't know if people will not change the matrix externally. copy has to be kept

Reply:
It is just strange to not change it in place when we call update, but change it in place when we call setWorldMatrix.
I mean, if one calls `setWorldMatrix` after an update, the matrix that will be modified will be one the one passed to the `update` function, which is the world Matrix of the Mesh. Do we really want to change the world Matrix of the change in this case ? (see our previous discussion here https://github.com/BabylonJS/Babylon.js/pull/5219#issuecomment-423580302)

Reply:
Actually this function should be removed IMHO. Having an accessor to it for debugging purpose is fine, but why a setter ? The only way to change it should to call `reConstruct` or `update`, as it would also update all the world coordinates stuff.

I guess some API change is allowed for a major version release such as 4.0, isn't it ? 😉